src/downloading/dumpFigureLibraries.ts

```diff
@@ -29,25 +29,51 @@ export async function dumpFigureLibraries(
       await fs.mkdir(resolvedOutPath, { recursive: true });
 
       const swfLocation = path.join(resolvedOutPath, fileName);
-
-      const success = () =>
-        dispatch({ type: "FIGURE_ASSETS_PROGRESS_SUCCESS", payload: fileName });
+      const downloadSuccess = () =>
+        dispatch({ type: "FIGURE_ASSETS_DOWNLOAD_COUNT" });
 
       try {
         await fs.stat(swfLocation);
-        success();
-        return;
+        return downloadSuccess();
+      } catch (e) {}
+
+      try {
+        const buffer = await tryFetchBuffer(file);
+        await fs.writeFile(swfLocation, buffer);
+
+        downloadSuccess();
       } catch (e) {
-        // Continue if file doesnt exist yet
+        console.log("dumpFigureLibraries error: ", e);
       }
+    },
+    { concurrency: Infinity }
```

Comment:
Is this `Infinity` intentional? I believe when I ran it the servers instantly blocked me because of it.

Reply:
Yes, both are intentional. I let this as `Infinity` because you can download as many file as you are able. The `retryFetch` has a internal watcher which will check if the file have been downloaded or not. If the download is stucked by Sulake CDN then the watcher will abort the request and retry to download it automatically.

Reply:
Well, did another test today and it seems to be stucked. Yesterday I had no issues while downloading these files with the `Infinity` as concurrency limit. I'll revert it to 30.